egg/core/callbacks.py

```diff
@@ -44,11 +43,22 @@ def on_epoch_begin(self, epoch: int):
     def on_epoch_end(self, loss: float, logs: Interaction, epoch: int):
         pass
 
+    def on_batch(self, logs: Interaction, loss: float, data_len: int):
+        pass
+
 
 class ConsoleLogger(Callback):
-    def __init__(self, print_train_loss=False, as_json=False):
+    def __init__(self, print_train_loss=False, as_json=False, print_batch_progress=False, n_epochs=-1):
```

Comment:
Reasons to start from -1? And I'm thinking ConsoleLogger and ProgressBarUpdater should be two separate callbacks